keel-core/src/main/kotlin/com/netflix/spinnaker/keel/validators/DeliveryConfigValidator.kt

```diff
@@ -83,5 +86,20 @@ class DeliveryConfigValidator {
         }
       }
     }
+
+    /**
+     * check: all referenced artifacts exist
+     */
+    config.environments.forEach { environment ->
+      environment.resources.forEach { resource ->
+        (resource.spec as? ArtifactReferenceProvider)
+          ?.artifactReference
+          ?.also {
+            if (!refs.contains(it)) {
+              throw NonexistentArtifactReferenceException(it, refs)
+            }
+          }
+      }
+    }
```

Comment:
I know in the vast majority of cases there will only be a single artifact, but is there a danger of us getting into whack-a-mole problems with this class in general? Fix one problem, a different exception pops up, fix that, yet another pops up. I wonder if there's a way we can group all the validation results at once (in a future PR -- I'm not suggesting doing it here).